### PR TITLE
Add new compute-region-network-firewall-policy-with-rules resource

### PR DESCRIPTION
**Pull request by mihhalj**
Add a new resource to allow updating all rules in firewall policy at once.

Fixes 
https://github.com/hashicorp/terraform-provider-google/issues/19195

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_compute_region_network_firewall_policy_with_rules` (beta)
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/compute_region_network_firewall_policy_with_rules_full.tf.erb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          name = "tagValues/${google_tags_tag_value.secure_tag_value_1.name}"
+        }
+      }
+      disabled = true
+    }
+}
+
+resource "google_network_security_address_group" "address_group_1" {
+  provider  = google-beta 
+  name        = "address-group-1"
```
Comment (BBBmau) [inline mmv1/templates/terraform/examples/compute_region_network_firewall_policy_with_rules_full.tf.erb:57]: You'll want to have a test name for security address_group, this is resulting in the failing test below:

```hcl
    vcr_utils.go:167: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating AddressGroup: googleapi: Error 409: Resource 'projects/ci-test-project-188019/locations/us-west2/addressGroups/address-group-1' already exists
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.ResourceInfo",
            "resourceName": "projects/ci-test-project-188019/locations/us-west2/addressGroups/address-group-1"
          }
        ]
        
          with google_network_security_address_group.address_group_1,
          on terraform_plugin_test.tf line 56, in resource "google_network_security_address_group" "address_group_1":
          56: resource "google_network_security_address_group" "address_group_1" {
        
        
        Error: Error waiting to create TagKey: Error waiting for Creating TagKey: Error code 6, message: The TagKey namespace 'ci-test-project-188019/tag-key' already exists.
        
          with google_tags_tag_key.secure_tag_key_1,
          on terraform_plugin_test.tf line 67, in resource "google_tags_tag_key" "secure_tag_key_1":
```

Comment (mihhalj) [reply mmv1/templates/terraform/examples/compute_region_network_firewall_policy_with_rules_full.tf.erb]: Done - I fixed the tests.

**Comment by mihhalj** [comment]
Could I ask for review?

**Comment by mihhalj** [comment]
Friendly ping - it's the same change as in https://github.com/GoogleCloudPlatform/magic-modules/pull/11524 or https://github.com/GoogleCloudPlatform/magic-modules/pull/11525 which were already approved - this one applies to regional resource.

File: mmv1/templates/terraform/encoders/resource_compute_region_network_firewall_policy_with_rules.go.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+delete(obj, "rules") // Rules are not supported in the create API
```
Comment (BBBmau) [inline mmv1/templates/terraform/encoders/resource_compute_region_network_firewall_policy_with_rules.go.erb:1]: Could you provide a link to where the Rules is not supported by the create API? I've been reviewing the PR and noticed that the `insert` for adding rules contains `rules` as part of the request body. Maybe I'm not referencing the correct resource?

https://cloud.google.com/compute/docs/reference/rest/v1/regionNetworkFirewallPolicies/insert

Comment (BBBmau) [reply mmv1/templates/terraform/encoders/resource_compute_region_network_firewall_policy_with_rules.go.erb:1]: thanks for clarification!

**Review by BBBmau — APPROVED** [review]
everything else is good, again thanks for the clarification that was the only thing left in this review!

**Pull request merged**